kubeadm unit test requires root k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade: TestEnforceRequirements/Fail_pre-flight_check

### What happened?

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/92316/pull-kubernetes-unit-experimental/1610708038858051584

```
{Failed  === RUN   TestEnforceRequirements/Fail_pre-flight_check
[upgrade/config] Making sure the configuration is correct:
[preflight] Running pre-flight checks.
    common_test.go:113: enforceRequirements returned unexpected error, expected: ERROR CoreDNSUnsupportedPlugins, got [preflight] Some fatal errors occurred:
        	[ERROR IsPrivilegedUser]: user is not running as root
        [preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
    --- FAIL: TestEnforceRequirements/Fail_pre-flight_check (0.00s)
}
```

### What did you expect to happen?

_unit_ tests should run without root.

This test should be refactored or otherwise removed from `make test`. `make test` should pass as an ordinary user. https://github.com/kubernetes/kubernetes/issues/99881

### How can we reproduce it (as minimally and precisely as possible)?

run `cmd/kubeadm/app/cmd/upgrade` tests as non-root

--- a/cmd/kubeadm/app/cmd/upgrade/common_test.go
+++ b/cmd/kubeadm/app/cmd/upgrade/common_test.go
@@ -72,18 +72,20 @@ func TestEnforceRequirements(t *testing.T) {
 	}
 
 	tcases := []struct {
-		name          string
-		newK8sVersion string
-		dryRun        bool
-		flags         applyPlanFlags
-		expectedErr   string
+		name               string
+		newK8sVersion      string
+		dryRun             bool
+		flags              applyPlanFlags
+		expectedErr        string
+		expectedErrNonRoot string
 	}{
 		{
 			name: "Fail pre-flight check",
 			flags: applyPlanFlags{
 				kubeConfigPath: fullPath,
 			},
-			expectedErr: "ERROR CoreDNSUnsupportedPlugins",
+			expectedErr:        "ERROR CoreDNSUnsupportedPlugins",
+			expectedErrNonRoot: "user is not running as", // user is not running as (root || administrator)
 		},
 		{
 			name: "Bogus preflight check specify all with individual check",
@@ -109,8 +111,13 @@ func TestEnforceRequirements(t *testing.T) {
 				t.Error("Expected error, but got success")
 			}
 
-			if err != nil && !strings.Contains(err.Error(), tt.expectedErr) {
-				t.Fatalf("enforceRequirements returned unexpected error, expected: %s, got %v", tt.expectedErr, err)
+			expErr := tt.expectedErr
+			// pre-flight check expects the user to be root, so the root and non-root should hit different errors
+			if os.Getuid() != 0 && len(tt.expectedErrNonRoot) != 0 {
+				expErr = tt.expectedErrNonRoot
+			}
+			if err != nil && !strings.Contains(err.Error(), expErr) {
+				t.Fatalf("enforceRequirements returned unexpected error, expected: %s, got %v", expErr, err)
 			}
 
 		})

